### ISSUE TITLE
aria-allowed-role fails for role spinbutton on input type tel

### ISSUE THREAD

**Issue by erin-doyle** [dequelabs/axe-core#1928]
Expectation: `<input type="tel" role="spinbutton" />` should pass aria-allowed-role

Actual: It fails.  It looks like maybe only `type="text"` and `type="number"` pass.

Motivation: Inputs of `type="tel"` are used frequently for inputs that need to take numerical values, including period and comma characters for decimal values.  It seems to me that this should be considered a valid type for a `role="spinbutton"`.

**Comment by WilcoFiers** [comment]
Howdy. This isn't valid according to [HTML in ARIA](https://www.w3.org/TR/html-aria/#el-input-tel), which is what this rule is based on. But I see your point, and I'm not sure there are any accessibility issues that could come from it. I'm in favour of changing this. Would you be interested in putting together a pull request?

**Comment by erin-doyle** [comment]
Yep I can put together a pull request for this.  Thanks!

**Comment by straker** [comment]
@erin-doyle Hello Erin, any update on creating a pr? Happy to help in any way.

**Comment by erin-doyle** [comment]
@straker I hit a really busy streak, I'll see if I can knock this out this week, shouldn't take long if I can just carve out the time!

**Comment by erin-doyle** [comment]
I'm starting to work on this finally and I've added a test to test/checks/aria/aria-allowed-role.js and expected that just by adding to the following lib/commons/aria/index.js it would work but that doesn't seem to be enough:

```
spinbutton: {
		type: 'widget',
		attributes: {
			allowed: [
				'aria-valuetext',
				'aria-required',
				'aria-readonly',
				'aria-errormessage',
				'aria-valuemax',
				'aria-valuemin'
			],
			required: ['aria-valuenow']
		},
		owned: null,
		nameFrom: ['author'],
		context: null,
		implicit: ['input[type="number"]'],
		unsupported: false,
		allowedElements: {
			nodeName: 'input',
			properties: {
				type: ['text', 'tel'] <-- added 'tel' here
			}
		}
	}
```

So I'll keep looking for where else I must need to make a change but if you see this first and can point me in the right direction that'd be great!

**Comment by straker** [comment]
You're on the right track! The problem is that we call into `lookupTable.evaluateRoleForElement` as the final check in `aria.isAriaRoleAllowedOnElement`, and we default `input[type=tel]` to [combobox](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/aria/index.js#L2343-L2347). Just change that to be like the text (minus searchbox) one above and it should work.

**Comment by erin-doyle** [comment]
Awesome, got it, thanks!  PR posted!

**Comment by straker** [comment]
We like to leave the tickets open and make sure we get them qa'd and docs updated before closing. But thanks for being proactive!

**Comment by erin-doyle** [comment]
Oops sorry about that!  Was just trying to keep Issues cleaned up!  Makes sense though, thanks!

**Comment by padmavemulapati** [comment]
Verified for 'aria-allowed-role' rule, its not giving any failure and thr rule is appearing in passes. 
Which is fix and expected.

![image](https://user-images.githubusercontent.com/47409579/76588433-0587fd80-650d-11ea-93ab-39ab15ca78ff.png)

**Comment by straker** [comment]
No docs needed
